sidechain/consensus/aura/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+//! Aura worker for the sidechain.
+//!
+//! It is inspired by parity's implementation but has ben greatly amended.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, feature(assert_matches))]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+#[macro_use]
```

Comment:
so do we still need the `#[macro_use]`? or is this just a copy-paste legacy?

Reply:
It is somewhat legacy. Before, this was the only way to get macros. Now it is not strictly necessary anymore, could also import `std::fmt::format` to get what we want.

Reply:
Okay, but that means there is a real benefit from using that annotation, it's not just dead code?